sdk/storage/azblob/blockblob/client.go

```diff
@@ -165,6 +165,20 @@ func (bb *Client) Upload(ctx context.Context, body io.ReadSeekCloser, options *U
 	return resp, err
 }
 
+// UploadBlobFromURL - The Put Blob from URL operation creates a new Block Blob where the contents of the blob are read from
+// a given URL. Partial updates are not supported with Put Blob from URL; the content of an existing blob is overwritten
+// with the content of the new blob. To perform partial updates to a block blob’s contents using a source URL, use the Put
+// Block from URL API in conjunction with Put Block List.
+// For more information, see https://learn.microsoft.com/rest/api/storageservices/put-blob-from-url
+func (bb *Client) UploadBlobFromURL(ctx context.Context, copySource string, options *UploadBlobFromURLOptions) (UploadBlobFromURLResponse, error) {
+	opts, httpHeaders, leaseAccessConditions, cpkInfo, cpkSourceInfo, modifiedAccessConditions, sourceModifiedConditions := options.format()
+
+	resp, err := bb.generated().PutBlobFromURL(ctx, int64(0), copySource, opts, httpHeaders, leaseAccessConditions,
```

Comment:
we might need trasacational md5/crc or setting up md5 sum feature here as well.

Reply:
In this case, we will need to follow the same pattern as https://github.com/Azure/azure-sdk-for-go/pull/20356.  Keep in mind the public interface needs to be future proofed, as we will be adding the ability for the SDK to calculate checksums in the future

Reply:
We just need to support `x-ms-source-content-md5`, and it will not be possible for the SDK to pre-calculate this checksum.

https://learn.microsoft.com/en-us/rest/api/storageservices/put-blob-from-url?tabs=azure-ad#request-headers

Reply:
Hold off on this for now, we need to figure out what we are doing with checksuming in AzCopy.  Consider this comment non-blocking.